### PR TITLE
Add Checks for Operation Definition in GraphQL Query

### PR DESCRIPTION
**Pull request by desout**
This PR addresses missing checks for operation definitions in GraphQL queries.

Changes include:

1. Added condition to check if definition name is present then insert it into the props vector.
2. For cases where no operation type is provided, it defaults to "query".
3. Added logic to create all_definitions vector, and a check for presence of more than one definition. In case where a name is absent for any of these definitions, an appropriate error is thrown.

Lay the groundwork for further validation checks in GraphQL query processing based on the behavior of 'gajus/babel-plugin-graphql-tag' project.

While extensive testing has been done locally, it has not been fully covered by automated tests due to Rust experience constraints. Specifically, error/panic scenarios demand more test coverage.

This effort has mimicked necessary logic from an existing solution found at https://github.com/gajus/babel-plugin-graphql-tag/blob/master/test/fixtures/graphql-tag. The current changes seek to ensure similar, comprehensive validation checks in Rust implementation.
#21

### CODE REVIEW

**Comment by desout** [comment]
@rishabh3112  Hello, could you, please, take a look on this changes?

**Comment by rishabh3112** [comment]
@desout Thanks for PR! Left few comments.

**Comment by desout** [comment]
@rishabh3112 Hello, Where i can check your comments? :)

File: transforms/graphql_tag/src/parser/utils.rs [hunk 1]
```diff
@@ -10,6 +10,10 @@ pub fn get_key_value_node(key: String, value: Expr) -> PropOrSpread {
 }
 
 pub fn get_operation_token(operation_type: Option<OperationType>) -> String {
+    if operation_type == None {
```
[the remark this answers is not in the document]
Comment (desout) [reply transforms/graphql_tag/src/parser/utils.rs]: done

File: transforms/graphql_tag/src/parser/nodes/definitions/operation.rs [hunk 2]
```diff
@@ -34,9 +31,16 @@ pub fn create_operation_definition(definition: OperationDefinition, span: Span)
 
     let mut opr_def = ObjectLit {
         span,
-        props: vec![kind, name, directives, variable_definitions, operation],
+        props: vec![kind, directives, variable_definitions, operation],
     };
 
+    if definition.name() != None {
```
[the remark this answers is not in the document]
Comment (desout) [reply transforms/graphql_tag/src/parser/nodes/definitions/operation.rs]: done

File: transforms/graphql_tag/src/parser/nodes/definitions/mod.rs [hunk 3]
```diff
@@ -42,6 +42,15 @@ pub fn create_definition(definition: Definition, span: Span) -> Option<ExprOrSpr
...
     let mut all_definitions = vec![];
+    let definitions_length = definitions.clone().count();
+
+    if definitions_length > 1 && definitions.clone().filter(|def| match def {
+        Definition::OperationDefinition(_) => true,
+        _ => false
+    }).any(|def| def.name() == None) {
+        panic!("GraphQL query must have name.");
+    }
+
```
[the remark this answers is not in the document]
Comment (desout) [reply transforms/graphql_tag/src/parser/nodes/definitions/mod.rs]: done

**Comment by desout** [comment]
@rishabh3112 Please review

**Comment by rishabh3112** [comment]
@desout will check and merge this weekend, thanks for contributions! Also, can you please change commits by using same email address as you have with github account? (for clearer commit history)

**Comment by desout** [comment]
@rishabh3112 Updated commits with proper email

**Review by rishabh3112 — APPROVED** [review]

**Pull request merged**